packages/context/src/inject.ts

```diff
@@ -165,29 +187,109 @@ export namespace inject {
   ) {
     return inject(bindingKey, metadata, resolveAsSetter);
   };
+
+  /**
+   * Inject an option from `options` of the parent binding. If no corresponding
+   * option value is present, `undefined` will be injected.
+   *
+   * @example
+   * ```ts
+   * class Store {
+   *   constructor(
+   *     @inject.options('x') public optionX: number,
+   *     @inject.options('y') public optionY: string,
+   *   ) { }
+   * }
+   *
+   * ctx.bind('store1').toClass(Store).withOptions({ x: 1, y: 'a' });
+   * ctx.bind('store2').toClass(Store).withOptions({ x: 2, y: 'b' });
+   *
+   *  const store1 = ctx.getSync('store1');
+   *  expect(store1.optionX).to.eql(1);
+   *  expect(store1.optionY).to.eql('a');
+
+   * const store2 = ctx.getSync('store2');
+   * expect(store2.optionX).to.eql(2);
+   * expect(store2.optionY).to.eql('b');
+   * ```
+   *
+   * @param bindingKey Optional property path of the option. If is `''` or not
+   * present, the `options` object will be returned.
+   * @param metadata Optional metadata to help the injection
+   */
+  export const options = function injectOptions(
+    bindingKey?: string,
+    metadata?: Object,
+  ) {
+    return inject(bindingKey || '', metadata, resolveAsOptions);
+  };
+
+  /**
+   * Inject an array of values by a tag
+   * @param bindingTag Tag name or regex
+   * @example
+   * ```ts
+   * class AuthenticationManager {
+   *   constructor(
+   *     @inject.tag('authentication.strategy') public strategies: Strategy[],
+   *   ) { }
+   * }
+   * ```
+   */
+  export const tag = function injectTag(bindingTag: string | RegExp) {
+    return inject('', {tag: bindingTag}, resolveByTag);
+  };
 }
 
 function resolveAsGetter(ctx: Context, injection: Injection) {
+  // No resolution session should be propagated into the getter
```

Comment:
I initially excluded the session from the getter because the session state will change for the current chain. Now I fixed it by cloning the session for getter.